tests/testthat/test-linear_pool.R

```diff
@@ -409,3 +409,114 @@ test_that("(weighted) quantiles correctly calculated - lognormal family", {
+                         sample_outputs$output_type_id == 1] <-
+    c(100, 325, 400, 300)
+  sample_outputs$value[sample_outputs$location == "888" &
+                         sample_outputs$output_type_id == 2] <-
+    c(250, 350, 500, 250)
+  sample_outputs$value[sample_outputs$location == "888" &
+                         sample_outputs$output_type_id == 3] <-
+    c(150, 300, 500, 350)
+
+  expected_outputs <- sample_outputs |>
+    dplyr::mutate(
+      output_type_id = paste0(.data[["model_id"]], .data[["output_type_id"]]),
+      model_id = "hub-ensemble"
+    ) |>
+    hubUtils::as_model_out_tbl()
+  actual_outputs <- sample_outputs |>
+    linear_pool(
+      weights = NULL,
+      task_id_cols = c("target_date", "target", "horizon", "location"),
+      output_samples = NULL,
+      numeric_output_type_ids = FALSE
+    )
+
+  expect_equal(actual_outputs, expected_outputs)
+})
+
+
+test_that("ensemble of samples is correctly calculated for more complex cases", {
+  sample_outputs <- expand.grid(stringsAsFactors = FALSE,
+                                model_id = letters[1:4],
+                                location = c("222", "888"),
+                                horizon = 1, #week
+                                target = "inc death",
+                                target_date = as.Date("2021-12-25"),
+                                output_type = "sample",
+                                output_type_id = 1:3,
+                                value = NA_real_)
+
+  sample_outputs$value[sample_outputs$location == "222" &
+                         sample_outputs$output_type_id == 1] <-
+    c(40, 30, 45, 80)
+  sample_outputs$value[sample_outputs$location == "222" &
+                         sample_outputs$output_type_id == 2] <-
+    c(60, 40, 75, 20)
+  sample_outputs$value[sample_outputs$location == "222" &
+                         sample_outputs$output_type_id == 3] <-
+    c(10, 70, 15, 50)
+  sample_outputs$value[sample_outputs$location == "888" &
+                         sample_outputs$output_type_id == 1] <-
+    c(100, 325, 400, 300)
+  sample_outputs$value[sample_outputs$location == "888" &
+                         sample_outputs$output_type_id == 2] <-
+    c(250, 350, 500, 250)
+  sample_outputs$value[sample_outputs$location == "888" &
+                         sample_outputs$output_type_id == 3] <-
+    c(150, 300, 500, 350)
+
+  fweight <- data.frame(model_id = letters[1:4], weight = 0.1 * (1:4))
+
+  reindexed_outputs <- sample_outputs |>
+    dplyr::mutate(output_type_id = paste0(.data[["model_id"]], .data[["output_type_id"]]))
+  expected_outputs <- reindexed_outputs |>
+    dplyr::filter(output_type_id %in% c("a2", "a3", "b1", "d1", "d3"))
+  for (i in 1:2) {
+    expected_outputs <- reindexed_outputs |>
+      dplyr::filter(.data[["model_id"]] %in% letters[(i + 1):4]) |>
+      dplyr::mutate(output_type_id = paste0(.data[["output_type_id"]], i)) |>
+      dplyr::bind_rows(expected_outputs)
+  }
```

Comment:
Where did these expectations come from?  It's not immediately clear to me why these are "correct" results or would be expected a priori.